### PR TITLE
Dockerize vulnerablecode

### PR DESCRIPTION
**Pull request by sbs2001**
Fixes https://github.com/nexB/vulnerablecode/issues/229

Signed-off-by: Shivam Sandbhor <shivam.sandbhor@gmail.com>

### CODE REVIEW

**Comment by sbs2001** [comment]
This is a repaste of the issue I am facing from gitter :

```
All works fine if the project is ran using docker-compose but I want the project to work without docker too, for that to happen I have specify "HOST":"localhost" in the settings.py manually which is not ideal. How should I tell django where to look for db depending upon whether it's ran in a container or normally ? I'm thinking of using an env variable in the docker-compose file and add some logic in settings.py to interpret this env variable. Is there a better way ?


```

File: README.md [hunk 1]
```diff
@@ -12,134 +15,404 @@
...
+  
+
+  
+
 VulnerableCode is a FOSS database of vulnerabilities and the FOSS packages they impact. It is made by the FOSS community to improve and secure the open source software ecosystem.
 
+  
+
+  
+
```
Comment (pombredanne) [inline README.md:33]: Why all the new blank lines?

File: README.md [hunk 2]
```diff
@@ -12,134 +15,404 @@
...
+- Docker Engine. Find instructions to install it [here](https://docs.docker.com/get-docker/)
+
+- Docker Compose. Find instructions to install it [here](https://docs.docker.com/compose/install/#install-compose)
+
+  
+
+Use `sudo docker-compose up` to start VulnerableCode.
```
Comment (pombredanne) [inline README.md:129]: Do we want to promote using sudo there?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+version: '3'
```
Comment (pombredanne) [inline docker-compose.yml:1]: what the verion there? the one for the compose file?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FROM python:3.8
+ENV PYTHONUNBUFFERED 1
```
Comment (pombredanne) [inline Dockerfile:2]: Can you explain why this with a comment?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+RUN mkdir /vulnerablecode
+WORKDIR /vulnerablecode
+ADD . /vulnerablecode/
+RUN pip install -r requirements.txt
```
Comment (pombredanne) [inline Dockerfile:6]: Can we add some labels to the image such that we know its base image, origin and license, author, maintainers, etc? (See how complicated a Docker image analysis can be with https://github.com/nexB/scancode.io/blob/main/scanpipe/pipelines/docker.py )

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks!
LGTM except for some weird spacing and a suggestion on Docker image.

File: README.md (hunk 2 above)
Comment (sbs2001) [reply README.md:129]: Without `sudo` I get 
```
shivam@shivam-UX430UAR:~/coding/development/vulnerablecode$ docker-compose up
ERROR: Couldn't connect to Docker daemon - you might need to run `docker-machine start default`.
```

File: docker-compose.yml (hunk 3 above)
Comment (sbs2001) [reply docker-compose.yml:1]: yes. They have that all over in their docs.

File: README.md (hunk 1 above)
Comment (sbs2001) [reply README.md]: Sorry. Can you check now ?

File: Dockerfile (hunk 4 above)
Comment (sbs2001) [reply Dockerfile:2]: https://stackoverflow.com/questions/59812009/what-is-the-use-of-pythonunbuffered-in-docker-file . 

I'll add comments there

File: README.md [hunk 6]
```diff
@@ -38,40 +38,45 @@ The web interface enables community curation of data by enabling addition of new
...
+git clone https://github.com/nexB/vulnerablecode.git
+cd vulnerablecode
 ```
 
-### System requirements
+### Using Docker Compose
+The easiest way to set up VulnerableCode is by using the docker containers.
+For this you need to have the following installed.
+- Docker Engine. Find instructions to install it here
+- Docker Compose. Find instructions to install it here
```
Comment (pombredanne) [inline README.md:54]: What about this:
```suggestion
An easy way to set up VulnerableCode is with docker containers and docker compose.
For this you need to have the following installed.
- Docker Engine at https://docs.docker.com/engine/install/
- Docker Compose at https://docs.docker.com/compose/install/
```

**Review by pombredanne — CHANGES_REQUESTED** [review]
See my suggestions inline. Note that at this stage we may want to start using ReST for the README and docs like for the other projects ... but this should another PR.

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+services:
+  web:
+    environment:
+        - DJANGO_DEV=1
+        - VC_DB_HOST=db
+    build: .
+    command: bash -c "python manage.py makemigrations && python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
```
Comment (tdruez) [inline docker-compose.yml:9]: Why the "web" service have to "makemigrations"?

Comment (sbs2001) [reply docker-compose.yml:9]: I'm not sure what alternative you are suggesting.

 `web` service has only the django code(not the db) hence I don't see any other way we can do migrations.

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+LABEL "base_image": "pkg:docker/python@sha256%3Ae9b7e3b4e9569808066c5901b8a9ad315a9f14ae8d3949ece22ae339fff2cad0"
+LABEL "dockerfile_url":  "https://github.com/nexB/vulnerablecode/blob/develop/Dockerfile"
+LABEL "homepage_url":  "https://github.com/nexB/vulnerablecode"
+LABEL "license": "Apache-2.0"
```
Comment (pombredanne) [inline Dockerfile:14]: This is awesome :)

**Review by pombredanne — APPROVED** [review]
LGTM++ :bow: 
Merging

**Pull request merged**